cachi2/core/package_managers/yarn_classic/workspaces.py

```diff
@@ -59,16 +58,9 @@ def _ensure_workspaces_are_well_formed(
             )
 
 
-def _get_workspace_paths(
-    workspaces_globs: list[str],
-    source_dir: RootedPath,
-) -> Iterable[Path]:
+def _get_workspace_paths(workspaces_globs: list[str], source_dir: RootedPath) -> list[Path]:
     """Resolve globs within source directory."""
-
-    def all_paths_matching(glob: str) -> Generator[Path, None, None]:
-        return (pth.resolve() for pth in source_dir.path.glob(glob))
-
-    return chain.from_iterable(map(all_paths_matching, workspaces_globs))
```

Comment:
This is a good catch! Alternatively the iterator could have been converted right here by wrapping `chain` call with a `list()`. I suggest retaining the helper, its purpose is to make it abundantly explicit what is happening. I'd say double comprehension makes it a bit less comprehensible.